tests/macros/utils.js

```diff
@@ -0,0 +1,121 @@
+/* jshint node: true, mocha: true, esversion: 6 */
+
+// Provides utilities that as a whole constitute the macro test framework.
+
+var sinon = require('sinon'),
+    kumascript = require('../..'),
+    // Only do this once, since it crawls the file system.
+    loader = new kumascript.loaders.FileLoader();
+
+function createMacroTestObject(name, done) {
+    var ctx = new kumascript.api.APIContext({
+            errors: [],
+            source: '',
+            loader: loader,
+            log: sinon.spy(),
+            env: {
+                locale: 'en-US',
+                url: 'https://developer.mozilla.org/'
+            },
+            autorequire: {
+                'mdn': 'MDN-Common',
+                'Page': 'DekiScript-Page',
+                'String': 'DekiScript-String',
+                'Uri': 'DekiScript-Uri',
+                'Web': 'DekiScript-Web',
+                'Wiki': 'DekiScript-Wiki'
+            },
+        }),
+        macro = {};
+
+    // Give the test-case writer access to the macro's globals (ctx).
+    // For example, "macro.ctx.env.locale" can be manipulated to something
+    // other than 'en-US' or "macro.ctx.wiki.getPage" can be mocked
+    // using "sinon.stub()" to avoid network calls.
```

Comment:
I don't know either, but I think you're right based on things like [JSDoc](http://usejsdoc.org/about-getting-started.html). Will do!